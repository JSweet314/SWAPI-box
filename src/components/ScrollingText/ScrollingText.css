.scrolling-text {
  background-color: #18212A;
  color: #fff;
}