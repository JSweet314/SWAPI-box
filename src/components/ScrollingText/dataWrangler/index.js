const scrollingTextDataWrangler = (movieData) => {
  if (!movieData) return;
  const openingCrawl = movieData.opening_crawl;
  const releaseDate = movieData.release_date;
  const title = movieData.title;
  return { openingCrawl, title, releaseDate };
};

export default scrollingTextDataWrangler;