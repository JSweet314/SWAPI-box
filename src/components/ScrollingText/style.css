.scrolling-text {
  display: flex;
  justify-content: center;
  color: orange;
  font-size: 10rem;
  height: 80%;
  overflow: hidden;
  perspective: 300px;
  width: 100vw;
}

.crawl {
  position: relative;
  top: -1000px;
  transform-origin: 50% 100%;
  animation: crawl 90s linear;
}

.instructions {
  position: absolute;
  z-index: 100;
  animation: 53s instructions;
  animation-fill-mode: forwards;
  visibility: hidden;
}

@keyframes crawl {
  0% {
    /* The element starts below the screen */
    top: 650px;
    /* Rotate the text 20 degrees but keep it close to the viewer */
    transform: rotateX(20deg) translateZ(0);
  }
  100% {
    /* This is a magic number, but using a big one to make sure the text is fully off the screen at the end */
    top: -6000px;
    /* Slightly increasing the rotation at the end and moving the text far away from the viewer */
    transform: rotateX(25deg) translateZ(-2000px);
  }
}

@keyframes instructions {
  99% {
    visibility: hidden;
  }
  100% {
    visibility: visible;
  }
}