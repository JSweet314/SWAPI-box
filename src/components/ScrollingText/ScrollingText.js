import React from 'react';
import './ScrollingText.css';
import PropTypes from 'prop-types';

const ScrollingText = ({openingCrawl, movieTitle, releaseDate}) => {
  const openingCrawlParagraphs = openingCrawl.map((paragraph, index) => {
    return <p key={index}>{paragraph}</p>
  });

  return (
    <aside className="scrolling-text">
      {openingCrawlParagraphs}
      <p>{movieTitle}</p>
      <p>{releaseDate}</p>
    </aside>
  );
};

ScrollingText.propTypes = {
  openingCrawl: PropTypes.array.isRequired,
  movieTitle: PropTypes.string.isRequired,
  releaseDate: PropTypes.string.isRequired
};

export default ScrollingText;