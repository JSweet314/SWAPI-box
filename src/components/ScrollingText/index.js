import React, { Component } from 'react';
import './style.css';
import fetchScrollingText from './apiCalls/fetchScrollingText';

class ScrollingText extends Component {
  constructor() {
    super();
    this.state = {
      openingCrawl: '',
      title: '',
      releaseDate: ''
    };
  }

  componentDidMount() {
    fetchScrollingText()
      .then(filmData => {
        this.setState({
          openingCrawl: filmData.openingCrawl,
          title: filmData.title,
          releaseDate: filmData.releaseDate
        });
      })
      .catch(error => alert(error));
  }

  render() {
    const {openingCrawl, title, releaseDate} = this.state;
    return (
      <aside className="scrolling-text">
        <pre>{openingCrawl}</pre>
        <p>{title}</p>
        <p>{releaseDate}</p>
      </aside>
    );
  }
}

export default ScrollingText;