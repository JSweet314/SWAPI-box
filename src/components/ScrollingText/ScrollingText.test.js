import React from 'react';
import ScrollingText from './ScrollingText';
import { shallow } from 'enzyme';

describe('ScrollingText', () => {
  it('should match a snapshot', () => {
    const wrapper = shallow(
      <ScrollingText openingCrawl="" movieTitle="" releaseDate=""/>
    );
    expect(wrapper).toMatchSnapshot();
  });
});