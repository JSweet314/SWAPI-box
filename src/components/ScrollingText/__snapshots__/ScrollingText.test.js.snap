// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollingText should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScrollingText
    movieTitle=""
    openingCrawl={Array []}
    releaseDate=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [],
        <p />,
        <p />,
      ],
      "className": "scrolling-text",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "p",
      },
    ],
    "type": "aside",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [],
          <p />,
          <p />,
        ],
        "className": "scrolling-text",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
      ],
      "type": "aside",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
